crates/llama-cpp-server/src/lib.rs

```diff
@@ -202,6 +210,53 @@ pub async fn create_completion(
     (stream, prompt_info)
 }
 
+pub async fn create_completion_and_chat(
+    completion_model: &LocalModelConfig,
+    chat_model: &LocalModelConfig,
+) -> (
+    Arc<dyn CompletionStream>,
+    Arc<dyn ChatCompletionStream>,
+    PromptInfo,
+) {
+    let chat_model_path = resolve_model_path(&chat_model.model_id).await;
+    let chat_template = resolve_prompt_info(&chat_model.model_id)
+        .await
+        .chat_template
+        .unwrap_or_else(|| panic!("Chat model requires specifying prompt template"));
+
+    let model_path = resolve_model_path(&completion_model.model_id).await;
+    let prompt_info = resolve_prompt_info(&completion_model.model_id).await;
+
+    let server = Arc::new(LlamaCppSupervisor::new(
+        "chat",
+        chat_model.num_gpu_layers,
+        false,
+        &chat_model_path,
+        chat_model.parallelism,
+        Some(chat_template),
+        chat_model.enable_fast_attention.unwrap_or_default(),
+        chat_model.context_size,
+    ));
+    server.start().await;
+
+    let chat = ChatCompletionServer::new_with_server(server.clone()).await;
+
+    let completion = if completion_model.model_id == chat_model.model_id {
```

Comment:
also compare other model settings (two LocalModelConfig shall be exactly same other than the `chat_template` field)